docs/src/main/asciidoc/cdi.adoc

```diff
@@ -0,0 +1,361 @@
+////
+This guide is maintained in the main Quarkus repository
+and pull requests should be submitted there:
+https://github.com/quarkusio/quarkus/tree/master/docs/src/main/asciidoc
+////
+= Quarkus - Introduction to Contexts and Dependency Injection
+
+include::./attributes.adoc[]
+:numbered:
+:sectnums:
+:sectnumlevels: 4
+:toc:
+
+In this guide we're going to describe the basic principles of the programming model used all over the Quarkus universe. 
+Quarkus programming model is based on the http://docs.jboss.org/cdi/spec/2.0/cdi-spec.html[Contexts and Dependency Injection for Java 2.0, window="_blank"] specification.
+However, a full CDI implementation is not used under the hood.
+Quarkus only implements a subset of the CDI features - see also <<cdi-reference.adoc#supported_features,the list of supported features>> and <<cdi-reference.adoc#limitations,the list of limitations>>.
+On the other hand, there are quite a few <<cdi-reference#nonstandard_features,non-standard features>> and <<cdi-reference.adoc#build_time_apis,Quarkus-specific APIs>>.
+We believe that our efforts will drive the innovation of the CDI specification towards the build-time oriented developer stacks in the future.
+
+== _Q: OK. Let's start simple. What is a bean?_
+
+A: Well, a bean is a _container-managed_ object that supports a set of basic services, such as injection of dependencies, lifecycle callbacks and interceptors.
+
+== _Q: Wait a minute. What does it mean container-managed?_
+
+A: Simply put, you don't control the lifecycle of the object instance directly.
+Instead, you can affect the lifecycle through the declarative means, such as annotations, configuration, etc.
+The container is the _environment_ where your application runs.
+It creates and destroys the instances of beans, associates the instances with a designated context, and injects them into other beans.
+
+== _Q: What is it good for?_
+
+A: An application developer can focus on the business logic rather than "where and how" to obtain a fully initialized component with all of its dependencies.
```

Comment:
```suggestion
A: An application developer can focus on the business logic rather than "where and how" of obtaining a fully initialized component with all of its dependencies.
```

Reply:
Hm, that does not sound good to me. What about _"An application developer can focus on the business logic **rather than finding out "where and how" to obtain** a fully initialized component with all of its dependencies."_?

Reply:
Sure, that sounds good to me